source/slang/slang-ir-spirv-snippet.cpp

```diff
@@ -60,8 +61,58 @@ RefPtr<SpvSnippet> SpvSnippet::parse(UnownedStringSlice definition)
                 mapInstNameToIndex[instName] = (int)snippet->instructions.getCount();
                 tokenReader.Read(Slang::Misc::TokenType::OpAssign);
             }
-            inst.opCode = (SpvWord)tokenReader.ReadInt();
+            SpvOp opCode;
+            switch (tokenReader.NextToken().Type)
+            {
+            case Slang::Misc::TokenType::IntLiteral:
+                opCode = (SpvOp)tokenReader.ReadInt();
+                break;
+            case Slang::Misc::TokenType::Identifier:
+            {
+                auto opName = tokenReader.ReadWord();
+                if(!lookupSpvOp(opName.getUnownedSlice(), opCode))
+                {
+                    throw Misc::TextFormatException(
+                        "Text parsing error: Unrecognized SPIR-V opcode: " + opName);
+                }
+                break;
+            }
+            default:
+                throw Misc::TextFormatException(
+                    "Text parsing error: SPIR-V intrinsics must begin with an integer or opcode name");
+            }
+            inst.opCode = opCode;
             bool insideOperandList = true;
+            const bool isExtInst = inst.opCode == SpvOpExtInst;
+            bool isGLSLstd450OpcodeAllowed = false;
+            auto readExtInstOpcode = [&]()
+            {
+                switch (tokenReader.NextToken().Type)
+                {
+                case Slang::Misc::TokenType::IntLiteral:
+                    return (SpvWord)tokenReader.ReadInt();
+                    break;
+                case Slang::Misc::TokenType::Identifier:
+                {
+                    if(isGLSLstd450OpcodeAllowed)
+                    {
+                        auto opName = tokenReader.ReadWord();
+                        GLSLstd450 glslOpcode;
+                        if(!lookupGLSLstd450(opName.getUnownedSlice(), glslOpcode))
+                        {
+                            throw Misc::TextFormatException(
+                                "Text parsing error: Unrecognized SPIR-V GLSLstd450 opcode: " + opName);
+                        }
+                        printf("BNBBB: %d\n", glslOpcode);
+                        return (SpvWord)glslOpcode;
+                    }
+                }
+                // fallthrough
+                default:
+                    throw Misc::TextFormatException(
```

Comment:
We need to turn this kind of exception throwing into an proper diagnostic at some point.